lib/ansible/modules/source_control/git.py

```diff
@@ -977,6 +1011,32 @@ def main():
     track_submodules = module.params['track_submodules']
 
     result.update(before=None)
+
+    if archive:
+        try:
+            os.makedirs(dest)
+        except OSError as e:
+            pass
+
+        if archive_file is None:
+            archive_file = "%s/%s.%s" % (dest, version, archive_fmt)
+
+        repo_name = repo.split("/")[-1].replace(".git", "")
+        # Git archive is not supported by all git servers, so
+        # we will first clone and perform git archive from local directory
+        cmd = "%s clone %s %s" %(git_path, repo, repo_name)
+        (rc, out, err) = module.run_command(cmd, cwd=dest)
+        if rc != 0:
+            module.fail_json(msg="Failed to clone source")
+        # Perform archive from local directory
+        cmd = "%s archive --format=%s --output=%s %s" \
+              % (git_path, archive_fmt, archive_file, version)
+        (rc, out, err) = module.run_command(cmd, cwd="%s/%s" % (dest,
+                                                                repo_name))
+        if rc != 0:
+            module.fail_json(msg="Failed to perform archive operation")
+        module.exit_json(**result)
```

Comment:
This isn't showing changed=True for me when is succeeds. In fact, I don't know if it will ever complete without changing something, since right now we're failing if the git archive command isn't successful and if the directory creation isn't successful. Maybe a nice addition would be to compare the checksums of the contents of the archive destination to see if they are different so changed could be False.